experimental/traffic-portal/nightwatch/globals/globals.ts

```diff
@@ -0,0 +1,35 @@
+/*
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {NightwatchGlobals} from "nightwatch";
+
+/**
+ * Defines the configuration used for the testing environment
+ */
+export interface GlobalConfig extends NightwatchGlobals {
+	adminPass: string;
+	adminUser: string;
+	trafficOpsURL: string;
+}
+const config = {
+	chrome_headless: {},
+	default: {
+		adminPass: "twelve12",
+		adminUser: "admin",
+		trafficOpsURL: "https://localhost:6443"
+	}
+};
+config.chrome_headless = config.default;
```

Comment:
idk if this is actually any better, but you could do this inline in the `config` literal as a `get` literal, fwiw. It's a little more convoluted, but it makes the typing correct - the exported `config` has the type:
```typescript
{
    chrome_headless: {},
    default: {
        adminPass: string,
        adminUser: string,
        trafficOpsUrl: string
    }
}
```
because `typeof config.default` is assignable to `{}` the assignment on this line succeeds, but anyone trying to access properties of `chrome_headless` won't be able to because its type has no properties. Although that doesn't appear to actually matter, for whatever reason.

Reply:
I suppose you could also just put `as typeof this.default` on the initialization of `chrome_headless`. I just hate using `as` so I always try to figure out how to not use it.

Reply:
It isn't, because you should _never_ reference this object yourself. It should be accessed through `browser.globals` since the type changes at runtime.

Reply:
> I suppose you could also just put `as typeof this.default` on the initialization of `chrome_headless`. I just hate using `as` so I always try to figure out how to not use it.

I agree, however with how the typings for nightwatch are currently setup I don't think this is possible.

Reply:
I suppose it doesn't matter since its an injected dependency and you can just import the type exported here.